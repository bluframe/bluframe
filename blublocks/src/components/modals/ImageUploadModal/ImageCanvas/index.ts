import { RefObject, useEffect, useRef } from "react"
import ImageCanvas from "./ImageCanvas"
import prepareComponent from "@bluframe/grapple/prepareComponent"

interface Props {
  imageDataUrl: string
}

export interface ComponentProps {
  canvasRef: RefObject<HTMLCanvasElement>
}

const X_COORDINATE = 0
const Y_COORDINATE = 0

export const usePrepareComponent = ({
  imageDataUrl
}: Props): ComponentProps => {
  const canvasRef = useRef<HTMLCanvasElement>(null)

  useEffect(() => {
    if (!canvasRef.current) {
      return
    }

    const canvas = canvasRef.current
    const context = canvas.getContext("2d")
    const imageObj = new Image()

    imageObj.onload = () => {
      canvas.width = imageObj.width
      canvas.height = imageObj.height
      context?.drawImage(
        imageObj,
        X_COORDINATE,
        Y_COORDINATE,
        imageObj.width,
        imageObj.height
      )
    }
    imageObj.src = imageDataUrl
  }, [imageDataUrl])

  return {
    canvasRef
  }
}

export default prepareComponent<Props, ComponentProps>(usePrepareComponent)(
  ImageCanvas
)
