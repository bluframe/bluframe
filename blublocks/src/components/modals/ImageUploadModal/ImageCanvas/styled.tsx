import styled from "styled-components"
import mediaQueries from "styles/mediaQueries"

export const Canvas = styled.canvas`
  height: auto;
  max-width: 100%;

  ${mediaQueries.tablet} {
    max-width: 300px;
  }
`
