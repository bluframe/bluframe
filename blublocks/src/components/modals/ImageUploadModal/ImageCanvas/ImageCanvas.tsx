import { Canvas } from "./styled"
import { ComponentProps } from "."
import React from "react"

const ImageCanvas = ({ canvasRef }: ComponentProps) => (
  <Canvas ref={canvasRef} />
)

export default ImageCanvas
