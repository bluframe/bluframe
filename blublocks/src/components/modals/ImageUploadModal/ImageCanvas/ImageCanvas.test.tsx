import ImageCanvas from "./ImageCanvas"
import React from "react"
import { render } from "tests"

describe("ImageCanvas", () => {
  it("renders correctly", () => {
    const ref = React.createRef<HTMLCanvasElement>()

    const { container } = render(<ImageCanvas canvasRef={ref} />)

    expect(container).toMatchSnapshot()
  })
})
