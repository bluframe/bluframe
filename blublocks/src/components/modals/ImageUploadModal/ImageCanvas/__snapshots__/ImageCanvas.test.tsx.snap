// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCanvas renders correctly 1`] = `
<div>
  <canvas
    class="sc-bdDqHD fEUNSC"
  />
</div>
`;
